app/controllers/students_controller.rb

```diff
@@ -29,8 +29,9 @@ def destroy
     end
     
     private
+    
     def load_student
-        @student = Student.find(params[:id])
+        @student = Student.find(params[:id]).decorate
```

Comment:
Do you need the `decorate` here?
I can see you're using `decorates_assigned :student` in this controller, which means that accessing it as `student` in the view should automatically give you the decorated object.

I've seen in the view you're using `@student.full_name`, which might be why you've opted to explicitly call decorate. If you instead just use `student.full_name`, you should be able to remove the `decorate` here. Can you give that a shot and let me know how you go?